internal/daemon/worker/auth_rotation_test.go

```diff
@@ -112,21 +114,27 @@ func TestRotationTicking(t *testing.T) {
 	// Make sure we have a failsafe in case the below loop never finds what it's
 	// looking for; at expiration the List will fail and we'll hit the Require,
 	// exiting
-	deadlineCtx, deadlineCtxCancel := context.WithTimeout(c.Context(), 10*time.Minute)
+	testTimeout := 10 * time.Minute
+	testDeadline, ok := t.Deadline()
+	if ok {
+		testTimeout = time.Until(testDeadline) - time.Second
+	}
+	deadlineCtx, deadlineCtxCancel := context.WithTimeout(c.Context(), testTimeout)
```

Comment:
Curious what the reasoning for this change in. If I'm understanding it right, we're allowing for the test to run until the remainder of the test timeout (if there is one) as opposed to the standard 10 minute timeout? Should we expect this test to ever take that long?

Reply:
Hm, fair, I wanted this to respect the test timeout if it's shorter than the 10 minute timeout, but yeah, we should probably only override it when it's smaller, not when it's bigger. I've had this test fail because I use a shorter test timeout locally. I'll push an update 👍🏻